### PR TITLE
Add DOM reconciler example

### PR DESCRIPTION
**Pull request by jchavarri**
- Creates a new level of folders in `examples`.
- Moves existing `lambda-term` example to a subfolder
- Creates new DOM reconciler example based on `js_of_ocaml`. I've shamelessly copied parts and comments from the `lambda-term` example 😄  To play with the example, run `esy build:dom` and open the `index.html` file inside the `examples/dom` folder.

Should the README be updated? I noticed many parts needed some tweaks, but otoh the final implementation is quite more verbose so the simplicity of the readme is also beneficial to not overwhelm readers.

### CODE REVIEW

**Comment by bryphe** [comment]
@jchavarri - this is awesome! Thank you for doing this!

> To play with the example, run esy build:dom and open the index.html file inside the examples/dom folder.

Saw the 'Hello World' - very cool 😄 

> Should the README be updated? I noticed many parts needed some tweaks, but otoh the final implementation is quite more verbose so the simplicity of the readme is also beneficial to not overwhelm readers.

Ideally, it makes sense to update the README. It feels a bit 'handwavy' at the moment, since it doesn't actually work. Maybe down the road there is a cleaner example we could pick up. Happy to help here if you'd like - you did the hard work of making it actually compile!

It looks the build is failing when it runs `esy b dune @examples` - I suspect this is due to the change in the directory structure - we may have to change the `alias` in the `dune` file at the root to point to the new location: 
```
(alias
    (name examples)
    (deps (alias examples/lambda-term/all)))
```

Thanks again for implementing this, @jchavarri !

File: package.json [hunk 1]
```diff
@@ -5,6 +5,7 @@
   "license": "MIT",
   "scripts": {
     "build:js": "esy b dune build test/Test.bc.js",
+    "build:dom": "refmterr dune build ./examples/dom/WebReconciler.bc.js",
```
Comment (bryphe) [inline package.json]: For this to work on Windows - I needed to change it to `esy b refmterr dune build ./examples/dom/WebReconciler.bc.js`

**Review by bryphe — CHANGES_REQUESTED** [review]
Just needs two things before we can merge:
- Green CI build (I think tweaking that dune path at the root should address this)
- Fix `esy build:dom` for Windows - it might even be worth adding it to the `.ci/esy-build-steps.yml`, just so we know it runs correctly on all platforms.

LMK if there is anything I can help with-

**Comment by jchavarri** [comment]
Thanks for the quick feedback @bryphe !

~~I have implemented the changes in https://github.com/bryphe/reason-reactify/pull/31/commits/e7d979725257a14333f1a67a291d72d7befecfd7, but I don't see CI being triggered. Do you know what could be going on? 🤔~~ Edit: now it's green! ✅

**Comment by bryphe** [comment]
Changes look great to me, @jchavarri ! Thanks again, this is awesome 💯 

Bringing it in now!

**Pull request merged**